apps/rewards/src/attestation/attestation.service.ts

```diff
@@ -0,0 +1,102 @@
+import { KomenciLoggerService } from '@app/komenci-logger'
+import { ContractKit } from '@celo/contractkit'
+import { Injectable } from '@nestjs/common'
+import { Cron, CronExpression } from '@nestjs/schedule'
+import { v4 as uuidv4 } from 'uuid'
+import { NotifiedBlockService } from '../blocks/notifiedBlock.service'
+import { fetchEvents } from '../utils/fetchEvents'
+import { Attestation } from './attestation.entity'
+import { AttestationRepository } from './attestation.repository'
+
+const NOTIFIED_BLOCK_KEY = 'attestation'
+
+@Injectable()
+export class AttestationService {
+  isRunning = false
+
+  constructor(
+    private readonly attestationRepository: AttestationRepository,
+    private readonly notifiedBlockService: NotifiedBlockService,
+    private readonly contractKit: ContractKit,
+    private readonly logger: KomenciLoggerService
+  ) {}
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  async fetchAttestations() {
```

Comment:
This logic looks similar to the `sendInviteRewards` method. We should create a generalized event fetcher with a callback that allows us to do different actions with the events.